### ISSUE TITLE
Spaced repetition scheduler is buggy and unreliable 

### ISSUE THREAD

**Issue by tfiwm** [remnoteio/remnote-issues#278]
**Describe the bug**
There is some sort of bug in the whole scheduling algorithm, which is hard to understand or explain. All I can see is the manifestation of that in the review intervals. The reviews are just **messed up** IMO. See the screenshots for some examples

**To Reproduce**

1. Just create bunch of cards a keep reviewing them for 3 weeks to 1 month. 
2. Then check the meta data of random cards from that list 

**Expected behavior**
Cards should get scheduled as per the selected scheduler algorithm

**Screenshots**
In the following screenshot, 

**1.** Why was there a review on Nov 6th when the previous review was on Nov 5th with a *solid* response? I expect the next review after Nov 5th to be at least 10 days later
**2.** Why is the *next practice time* is ( in 2 months) on Jan 12th, 2021, when the *last practice time* was on Nov 12, 2020 and the interval before the last practice was just 6 days

![image](https://user-images.githubusercontent.com/24322880/99492498-86ac5580-2922-11eb-8963-5488b284a233.png)

In the following screenshot, 

1. Why are there three reviews on Nov 5th?
2. Why is the *next repetition time* **in 10 months, on Sep 25th 20201** (that is crazy) when the *last practice* was on Nov 14th, 2020 (the last interval was just 8 days). The jump is from 8 days to 10 months!
![image](https://user-images.githubusercontent.com/24322880/99492557-a3488d80-2922-11eb-9697-272fc633f2c6.png)

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser : Google Chrome (50%) Version 86.0.4240.198 (Official Build) (64-bit)

**Comment by tfiwm** [comment]
Another such card getting very high '**next repetition time**' value 
![image](https://user-images.githubusercontent.com/24322880/99627128-8d050500-29e8-11eb-889a-6f14f9ffa76f.png)

**Comment by MClatter** [comment]
I was wondering why can't I make this rem work. :)
![weirdRem](https://user-images.githubusercontent.com/2999584/99896202-5d335700-2c9f-11eb-8b6d-12cfde727cb8.png)

**Comment by anishaaa1** [comment]
To add, there's some cards I've come across in the queue for which the Easy intervals are crazy long (my guess is this happens with some cards I click 'Forgot' on but I'd have to test more to be sure).
Side note: I'm also steering clear of the Early button because it fully deactivates the cards and while some show up maybe 20 - 30 days later, most just never show up.

![image](https://user-images.githubusercontent.com/68901589/99896554-0f245080-2cab-11eb-836a-0886a6d0cf18.png)

**Comment by tfiwm** [comment]
> I was wondering why can't I make this rem work. :)
> ![weirdRem](https://user-images.githubusercontent.com/2999584/99896202-5d335700-2c9f-11eb-8b6d-12cfde727cb8.png)

Unfortunately, I also found one such card after November update. But I am still hoping that is because the card was created before the 'possible November fix' and cards created after November update would never skip the review. Please share if you find any newly created card getting disabled.

**Comment by tfiwm** [comment]
> To add, there's some cards I've come across in the queue for which the Easy intervals are crazy long (my guess is this happens with some cards I click 'Forgot' on but I'd have to test more to be sure).
> Side note: I'm also steering clear of the Early button because it fully deactivates the cards and while some show up maybe 20 - 30 days later, most just never show up.
> 
> ![image](https://user-images.githubusercontent.com/68901589/99896554-0f245080-2cab-11eb-836a-0886a6d0cf18.png)

This is exactly the issue, thank you, now I am convinced that I am not the only one. Probably not many people are checking the status of their intervals. I also feel there is some sort of issue with 'Hard' and/or 'Easy' responses, most of the problems are with cards which were answered hard or easy at some point in the past. 
I am also trying my luck with the custom SM2 scheduler to see if that implementation is more reliable. I am just realizing that I just don't like Remnote's built-in scheduler, especially how it handles cards during learning phase

**Comment by anishaaa1** [comment]
Since this is a general bug report for spaced repetition bugs, I've recently come across a new one. Some rems show up multiple times in the queue in a single day despite being several days into review + having clicked 'Solid' on them. I've been able to identify only one such rem but I know that there are more.

Here's a screenshot:

![Screen Shot 2020-12-09 at 3 39 13 PM](https://user-images.githubusercontent.com/68901589/101625472-e2a35f00-3a34-11eb-9096-14c483f60482.png)

**Comment by MClatter** [comment]
> 
> 
> Since this is a general bug report for spaced repetition bugs, I've recently come across a new one. Some rems show up multiple times in the queue in a single day despite being several days into review + having clicked 'Solid' on them. I've been able to identify only one such rem but I know that there are more.
> 
> Here's a screenshot:
> 
> ![Screen Shot 2020-12-09 at 3 39 13 PM](https://user-images.githubusercontent.com/68901589/101625472-e2a35f00-3a34-11eb-9096-14c483f60482.png)

I had the same/similar problem yesterday, a rem appeared twice back to back. Unfortunately I wasn't in the mood to take any action and now I can't remember which one was it. :)

**Comment by claforte** [comment]
I noticed a similar problem with rems I had edited, i.e. when I made changes to the clozes, they showed up twice in the queue (annoyance #1) and as a result they got scheduled very far in the future (e.g. something like from 1 week -> 2 weeks -> 1.5 month)

**Comment by gabriel-at-gthb** [comment]
Most of my cards are repeating on a daily basis, or within a few days.   These are cards that have been marked not to return for weeks or months.   It is very frustrating.   Someone on the community q&a forum said most of their problems were cured by the December update.    I just started using Remnote in December.    I've tried the desktop version on a windows pc, and the browser versions on my PC, Ipad, and Iphone.    The rems in my que repeat FAR to soon on any device.
